apache/superset
# fix(pivot_table_v2): Pivot table cannot validly group rows and keep ordering

### SUMMARY
Fix and improve `value_a_z`, `value_z_a` sort order for pivot table
1.  work on any size of cols and rows in pivot table options, groups them
2. keep subtotals with their groups
3. keep subtotals on required top/bottom position
4. switches to a-z order in case of same numerical values

### TESTING INSTRUCTIONS
Create any pivot v2 table with
1. at least two rows (or/and to cols) and summable metric
5. switch on subtotals
6. try switch different orderings (all row/col groups should be consistent)
7. try switch subtotal position (all row/col groups should be consistent and subtotals should be at expected place)

### ADDITIONAL INFORMATION

Known issue - not tested for non monotonic summable aggregates with natural order (sum, count, avg).
It's because it collect raw subtotals sumarrizing internal values.
Some aggregate function (min, max for) could have unexpected behavior in sorting.
